integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/assertion/storage/azure/AdlsGen2TestAssertion.java

```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.it.cloudbreak.newway.assertion.storage.azure;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.parameter.storage.AdlsGen2CloudStorageV4Parameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.cluster.storage.location.StorageLocationV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.cluster.storage.location.StorageLocationV4Response;
+import com.sequenceiq.it.cloudbreak.newway.assertion.AssertionV2;
+import com.sequenceiq.it.cloudbreak.newway.dto.stack.StackTestDto;
+
+public class AdlsGen2TestAssertion {
+
+    private AdlsGen2TestAssertion() {
+    }
+
+    public static AssertionV2<StackTestDto> stackContainsAdlsGen2Properties() {
+        return (testContext, testDto, cloudbreakClient) -> {
+            AdlsGen2CloudStorageV4Parameters parameters = testDto.getResponse().getCluster().getCloudStorage().getAdlsGen2();
+            if (parameters == null) {
+                throw new IllegalArgumentException("AdlsGen2 parameters should not be null in response!");
```

Comment:
All these checks where an exception is thrown could be extracted to a well-named private method. The code would become lot readable.

Reply:
or maybe extracted to a util class's method which throws an IllegalArgumentException with a message on the incoming boolean result. after that, this util approach could be used in other cases in the future (and the already existing cases)

Reply:
I wouldn't bother with such complex constructs here. Simple private methods here like `checkNullParamaters` and `checkStorageLocationCount`